dll/win32/shell32/wine/shellpath.c

```diff
@@ -87,8 +87,8 @@ DoGetProductType(PNT_PRODUCT_TYPE ProductType)
         return FALSE;
 
     cbValue = sizeof(szValue);
-    error = RegQueryValueExW(hKey, L"ProductType", NULL, &dwType, (LPBYTE)szValue, &cbValue);
-    if (!error && dwType == REG_SZ)
+    error = RegGetValueW(hKey, NULL, L"ProductType", RRF_RT_REG_SZ, NULL, (PVOID)szValue, &cbValue);
```

Comment:
`RegGetValueW` is for Windows Server 2008 and later. That is not available in Server 2003.

Reply:
This function is S03sp1+, though it is simpler to use it on Vista+ only, which Shell32 is.